mlflow/pyfunc/__init__.py

```diff
@@ -780,6 +782,7 @@ def spark_udf(spark, model_uri, result_type="double"):
             artifact_uri=model_uri, output_path=local_tmpdir.path()
         )
         archive_path = SparkModelCache.add_local_model(spark, local_model_path)
+        model = Model.load(os.path.join(local_model_path, MLMODEL_FILE_NAME))
```

Comment:
I've realized that we use `model` in the predict function as well. It should not matter but it is confusing. Can you rename this to `model_metadata`?